test/adaptive_load/fake_plugins/fake_input_variable_setter/fake_input_variable_setter.cc

```diff
@@ -0,0 +1,70 @@
+#include "test/adaptive_load/fake_plugins/fake_input_variable_setter/fake_input_variable_setter.h"
+
+namespace Nighthawk {
+
+namespace {
+
+absl::Status StatusFromProtoRpcStatus(const google::rpc::Status& status_proto) {
+  return absl::Status(static_cast<absl::StatusCode>(status_proto.code()), status_proto.message());
```

Comment:
Could this throw? If so, who should handle that?

Reply:
This was interesting... It turns out that casting an integer to an enum outside the defined range doesn't throw but produces undefined behavior. And C++ has no maintainable way to check for the largest value in this enum -- absl::StatusCode has a sentinel value that they say not to use, and it won't work anyway because of the gap from 16 to 20 [0].

Everyone who uses this field is expected to use standard Google codes, maybe through constants like grpc::DATA_LOSS.

I'm adding a warning about the enum issue to the proto comment. Do you think that's enough?

A safer but stranger alternative would be to define this field as not supporting anything above kAuthenticated=16, not tracking new absl::StatusCode values as they are added.

[0] https://github.com/abseil/abseil-cpp/blob/fbf0fdab62ac6fcc72b935d505f6837e0884fdc3/absl/status/status.h#L159

Reply:
My 2p would be: If the input can be trusted I think the comment suffices, and we should probably mention that in a comment explicitly. Hopefully ubsan in CI will have our back here too. If the input can't be trusted, we may need to think about how we can ensure defined behavior.. maybe then the "strange" alternative is the better option.